src/web/routes/admin/auth/auth-router.ts

```diff
@@ -0,0 +1,40 @@
+import type { NextFunction } from 'express';
+import { Router } from 'express';
+
+import type { CheckAuthRequest, CheckAuthResponse } from './types';
+
+import { figmaAuthService } from '../../../../infrastructure/figma';
+import { checkUserFigmaAuthUseCase } from '../../../../usecases';
+
+export const authRouter = Router();
+
+/**
+ * Checks whether the given Atlassian admin is authorized to call Figma API.
+ */
+authRouter.get(
+	['/checkAuth'],
+	function (req: CheckAuthRequest, res: CheckAuthResponse, next: NextFunction) {
+		const { connectInstallation, atlassianUserId } = res.locals;
+
+		checkUserFigmaAuthUseCase
+			.execute(atlassianUserId, connectInstallation)
+			.then((authorized) => {
+				if (authorized) {
+					return res.send({ authorized });
+				}
+
+				const authorizationEndpoint =
+					figmaAuthService.createOAuth2AuthorizationRequest({
+						atlassianUserId,
+						connectInstallation,
+						redirectEndpoint: `figma/oauth/callback`,
+					});
+
+				return res.send({
+					authorized,
+					grant: { authorizationEndpoint },
```

Comment:
Just noting this comment from @Roystbeef https://github.com/atlassian-labs/figma-for-jira/pull/106#issuecomment-1775418684

and confirming the proposal is to add an email field here without affecting the other `checkAuth` route?

Reply:
This is correct. Unlike other endpoints, the `/admin` endpoints are specific to UI and can be changed according to its needs. Therefore, if we need to add a field, we should make this change only for `/admin/auth/checkAuth`.

@Roystbeef , please, let me know if you would like me to take care of this change.